feat: Add RequestItemsView for beneficiary item requests

BarreiroReSird opened the pull request:
Introduces the RequestItemsView composable screen, allowing beneficiaries to search, filter, and select items to request, with quantity controls and a submission button. Also updates BeneficiaryPortalView layout spacing and imports, and adds a missing import in QuickActionsSection.

BarreiroReSird committed:
feat: Add RequestItemsView for beneficiary item requests

Introduces the RequestItemsView composable screen, allowing beneficiaries to search, filter, and select items to request, with quantity controls and a submission button. Also updates BeneficiaryPortalView layout spacing and imports, and adds a missing import in QuickActionsSection.

DiogoMachado04 committed:
docs: Sequence diagram for chat support

Creating the sequence diagram for chat support

DiogoMachado04 committed:
docs: Small correction in sequence diagram

Correcting a small detail in the sequence diagram for adding a product

BarreiroReSird committed:
feat: Add QueuedRequestsView and category icons

Introduces the QueuedRequestsView composable for displaying pending employee requests, along with new vector drawable icons for request categories (basket, cutlery, mop, perfume). Also removes an unused deployment target selection from project configuration.

Basiiii commented on app/app/src/main/java/com/lojasocial/app/ui/beneficiaries/BeneficiaryPortalView.kt:
```diff
@@ -20,21 +26,18 @@ fun BeneficiaryPortalView(useAppLayout: Boolean = true) {
+            Spacer(modifier = Modifier.height(4.dp))
 
             GreetingSection(
                 name = "Carla",
```
Mete o valor do nome do utilizador através dos metadados.

Basiiii commented on app/app/src/main/java/com/lojasocial/app/ui/beneficiaries/QuickActionsSection.kt:
```diff
@@ -12,6 +12,7 @@ import androidx.compose.ui.unit.sp
 import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.text.font.FontWeight
 import com.lojasocial.app.ui.components.ActionCard
+import com.lojasocial.app.ui.theme.BrandOrange
```
Ao resolver os conflitos pode ser que não precises disto

Basiiii commented on app/app/src/main/java/com/lojasocial/app/ui/beneficiaries/RequestItemsView.kt:
```diff
@@ -0,0 +1,340 @@
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+
+// --- Modelos de Dados (Para UI) ---
```
Devias criar num ficheiro separado de Models

Basiiii commented on app/app/src/main/java/com/lojasocial/app/ui/beneficiaries/RequestItemsView.kt:
```diff
@@ -0,0 +1,340 @@
+    var quantity: Int = 0
+)
+
+// --- Cores Personalizadas (Baseadas na imagem) ---
```
Mete as cores no ficheiro de tema da app

Basiiii commented on app/app/src/main/java/com/lojasocial/app/ui/beneficiaries/RequestItemsView.kt:
```diff
@@ -0,0 +1,340 @@
+    onBackClick: () -> Unit = {},
+    onSubmitClick: () -> Unit = {}
+) {
+    // Estado simulado (num caso real viria do ViewModel)
```
Tens de implementar o ViewModel, não podes fazer simulação

Basiiii commented on app/app/src/main/java/com/lojasocial/app/ui/beneficiaries/RequestItemsView.kt:
```diff
@@ -0,0 +1,340 @@
+    val categories = listOf("Todos", "Alimentar", "Limpeza", "Higiene")
+    var selectedCategory by remember { mutableStateOf("Todos") }
+
+    // Lista de produtos baseada na tua imagem
```
Tira os comentários do Gemini "baseados na tua imagem"

Basiiii commented on app/app/src/main/java/com/lojasocial/app/ui/beneficiaries/RequestItemsView.kt:
```diff
@@ -0,0 +1,340 @@
+    // Lista de produtos baseada na tua imagem
+    val products = remember {
+        mutableStateListOf(
+            ProductUI(1, "Pacote Arroz 1Kg", "Alimentar", 25, 0),
```
Produtos devem vir da base de dados

Basiiii commented on app/app/src/main/java/com/lojasocial/app/ui/beneficiaries/RequestItemsView.kt:
```diff
@@ -0,0 +1,340 @@
+                .padding(paddingValues)
+                .fillMaxSize()
+        ) {
+            // --- 1. Cabeçalho de Limite (Status Bar) ---
```
Divide estas partes em funçoes diferentes, talvez até ficheiros separados (componentes)

Basiiii commented on app/app/src/main/java/com/lojasocial/app/ui/beneficiaries/RequestItemsView.kt:
```diff
@@ -0,0 +1,340 @@
+val LightBg = Color(0xFFF5F7FA)      // Fundo cinza claro
+val DisabledBtn = Color(0xFFD1D5DB)  // Cinza do botão desativado
+
+@OptIn(ExperimentalMaterial3Api::class)
```
Não uses coisas Experimental

Basiiii commented on app/app/src/main/java/com/lojasocial/app/ui/beneficiaries/RequestItemsView.kt:
```diff
@@ -0,0 +1,340 @@
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun MakeOrderScreen(
```
MakeOrderScreen mas o nome do ficheiro é RequestItemsView?

Basiiii commented on app/.idea/deploymentTargetSelector.xml:
Não faz sentido tirar isto do ficheiro

zealves99 committed:
docs: update atas e convocatórias

update dos documentos de atas e convocatórias

zealves99 committed:
Merge branch 'main' of https://github.com/Basiiii/LojaSocial-IPCA

Basiiii committed:
fix: fixed auth persistence bug

Basiiii committed:
fix: fixed the bottom navigation & padding on portal selection screen

DiogoMachado04 committed:
chore: Fix sequence diagram formatting

Fixed the sequence diagram formatting

BarreiroReSird committed:
feat: Add pending requests feature for employees

Introduced PendingRequestsRepository, domain model, UI state, ViewModel, and new composable views to display pending requests for employees. Replaced QueuedRequestsView with a new implementation using PendingRequest and modularized RequestItem and StatusBadge components. Updated DI module for repository injection and adjusted project configuration files.

BarreiroReSird committed:
refactor: Refactor pending requests feature package structure

Moved PendingRequests-related classes from 'employees' to 'requests' and 'requests/components' packages for better organization. Updated repository import paths and renamed AppModule to PendingRequestsModule to reflect the new structure.

Basiiii committed:
Merge pull request #31 from Basiiii/feat/queued-requests-view

feat: Add PendingRequestsView and category icons

BarreiroReSird committed:
Merge branch 'feat/request-items-view' of https://github.com/Basiiii/LojaSocial-IPCA into feat/request-items-view

BarreiroReSird committed:
refactor: Refactor and modularize request items UI for beneficiaries

Refactored the RequestItemsView to use a new ViewModel, UI state, and repository pattern for product management. Extracted UI components (category filters, product row, status header, search bar, top app bar, and bottom bar) into separate files for better modularity and maintainability. Introduced domain and UI models for products, updated color theme, and improved state handling and UI responsiveness.

BarreiroReSird commented:
@Basiiii
